src/inferenceql/publish/asciidoc.clj

```diff
@@ -1,13 +1,77 @@
 (ns inferenceql.publish.asciidoc
-  (:import [inferenceql.publish YellBlockProcessor]
+  (:import [java.util HashMap]
            [org.asciidoctor Asciidoctor$Factory]
-           [org.jsoup Jsoup])
-  (:require [com.stuartsierra.component :as component]))
+           [org.asciidoctor Options]
+           [org.asciidoctor SafeMode]
+           [org.asciidoctor.ast ContentModel]
+           [org.asciidoctor.ast Document]
+           [org.asciidoctor.extension BlockProcessor]
+           [org.asciidoctor.extension Contexts]
+           [org.asciidoctor.extension DocinfoProcessor]
+           [org.asciidoctor.extension LocationType])
+  (:require [clojure.data.json :as json]
+            [com.stuartsierra.component :as component]
+            [hiccup.core :as hiccup]))
 
-(defonce ^:private asciidoctor
+(defn docinfo-processor
+  [f & {:keys [location]}]
+  {:pre [(some? f)
+         (some? location)
+         (contains? #{LocationType/HEADER LocationType/FOOTER} location)]}
+  (let [config (doto (HashMap.)
+                 (.put "location" (.optionValue location)))]
+    (proxy [DocinfoProcessor] [config]
+      (process [^Document document]
+        (f this document)))))
+
+(defn add-stylesheet-processor
+  [url]
+  (docinfo-processor
+   (fn [_ document]
+     (when (.isBasebackend document "html")
```

Comment:
is the `nil` return value handled well? Looks fine above, I guess `process` would just return nothing. or should it return `document` and act as identity?